pkgs/build-support/writable-dir-wrapper/default.nix

```diff
@@ -0,0 +1,190 @@
+{
+  lib,
+  writeShellApplication,
+  xorg,
+}:
+/**
+  A wrapper for programs that require a writable directory to function, typically
+  games or proprietary software that expect read-only resources to be placed alongside
+  config files and state files (e.g. logs).
+
+  # Inputs
+
+  *`options`* (Attribute set)
+
+  : Set of options for the wrapper.
+
+    `name` (String)
+
+    : File name of the wrapper.
+
+   `version` (String, _optional_)
```

Comment:
A lot of the code paths for this argument also feel...weird. I'm assuming it's for some of the games mentioned above, but I'm not sure how applicable it is to anything else really ~~although that's probably not a good argument as this helper is already pretty niche lol~~. It's logic could be reimplemented decently easy by derivations that need it rather than always here

Regardless, this should also probably be renamed to something like `packageVersion`, as using a regular `version` argument like this leads me to believe I'm editing the version of the script itself, which isn't super cool